Automattic/wp-calypso
# Media: Implement WIP advanced image settings dialog

Related: #307

This pull request seeks to implement a minimal set of behavior related to advanced image settings editing. Specifically, the changes included add a new TinyMCE toolbar button to be shown in the inline toolbar for images which, when clicked, toggles the display of a `<Dialog />` component, currently rendered with dummy content.

![wip](https://cloud.githubusercontent.com/assets/1779930/12101335/13dd2886-b302-11e5-9838-1db096c7264f.gif)

__Implementation notes:__

I fully expect the structure of the Redux state and `<EditorMediaAdvanced />` component to need extensive revisions as this functionality is built, thus the lack of proper documentation and testing. That said, feel free to comment on the proposed structure as-is.

__Testing instructions:__

Confirm that the included behavior is only shown in development environments by temporarily setting `post-editor/media-advanced` to `false` in `config/development.json` and restarting `make run`.

1. Navigate to the [Calypso post editor](http://calypso.localhost:3000/post)
2. Select a site, if prompted
3. Insert an image to the post content
4. Click the image to select it
5. Note that...
 - If `post-editor/media-advanced` is `true`, an "Edit" icon is shown in the inline toolbar
 - If `post-editor/media-advanced` is `false`, an "Edit" icon is not shown in the inline toolbar and there are no errors in the developer tools console
6. Click the Edit button in the inline toolbar
7. Note that a dialog is shown with dummy content
